### ISSUE TITLE
Add version check to application startup

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#683]
_[Trac time 20090606 191616Z]_ After loading a file on windows, trying to use the menubar results in:

```
Traceback (most recent call last):
  File "c:\Documents and Settings\
ob\Desktop\peppy-0.0.0\peppy\menu.py", line 339, in OnMenuOpen
    action.updateOnDemand(self)
  File "c:\Documents and Settings\
ob\Desktop\peppy-0.0.0\peppy\actions\__init__.py", line 865, in updateOnDemand
    self.widget.SetItemLabel(name)
AttributeError: 'MenuItem' object has no attribute 'SetItemLabel'
```

A debug print seems to show that it either happens on the first action or in this one:

```
menu.py:338 UserAccelerators.OnMenuOpen: action=ApplySettingsSameMode
```

**Comment by robmcmullen** [comment]
_[Trac time 20090606 191843Z]_ It seems that it's only in a FundamentalMode; using the keyboard to switch to a blank mode causes the menu bar to work again.

**Comment by robmcmullen** [comment]
_[Trac time 20090606 194601Z]_ Worked around using:

```
            if hasattr(self.widget, 'SetItemLabel'):
                self.widget.SetItemLabel(name)
            elif hasattr(self.widget, 'SetText'):
                self.widget.SetText(name)
```

in OnDemandActionNameMixin, but why is this necessary only in ApplySettingsSameMode, and all the other on demand things work with SetItemLabel?

**Comment by robmcmullen** [comment]
_[Trac time 20090606 194920Z]_ Argh, maybe it's because this windows machine only has wxPython 2.8.4.2 on it?

**Comment by robmcmullen** [comment]
_[Trac time 20090607 141714Z]_ Yep, because of 2.8.4.2.  2.8.7.1 doesn't have this problem.

**Comment by robmcmullen** [comment]
_[Trac time 20090607 155122Z]_ [old svn rev2107] Fixed #683: added checkVersion to Peppy application object
- set 2.8.7 as the minimum working version
- added attemptOldWxPythonVersionFixup to add code that tries to work around old version problems
- changed main application script to return message instead of error traceback when wxPython 2.6 is found
